Sources_v3/Database/DatabaseSession_Tests.swift

```diff
@@ -118,4 +118,39 @@ class DatabaseSession_Tests: StressTestCase {
         let message = try XCTUnwrap(loadedMessage)
         XCTAssert(loadedChannel.latestMessages.contains(message))
     }
+    
+    func test_saveEvent_unreadCountFromEventPayloadIsApplied() throws {
+        let eventPayload = EventPayload<DefaultDataTypes>(eventType: .messageNew,
+                                                          connectionId: .unique,
+                                                          cid: .unique,
+                                                          currentUser: .dummy(userId: .unique,
+                                                                              role: .user,
+                                                                              unreadCount: nil),
+                                                          unreadCount: .dummy)
+
+        _ = try await {
+            database.write({ session in
+                try! session.saveEvent(payload: eventPayload)
+            }, completion: $0)
+        }
+        
+        // Load current user
+        let currentUser = database.viewContext.currentUser()
+        
+        // Assert unread count is taken from event payload
+        XCTAssertEqual(Int16(eventPayload.unreadCount!.messages), currentUser?.unreadMessagesCount)
+        XCTAssertEqual(Int16(eventPayload.unreadCount!.channels), currentUser?.unreadChannelsCount)
+    }
+    
+    func test_saveCurrentUserUnreadCount_failsIfThereIsNoCurrentUser() throws {
+        // Save user with known unread count to the db
+        let saveUnreadCountError = try await {
+            database.write({ session in
+                try session.saveCurrentUserUnreadCount(count: .dummy)
+            }, completion: $0)
+        }
+        
+        // Assert error is thrown
+        XCTAssertNotNil(saveUnreadCountError)
```

Comment:
You can do `XCTAssert(error is ClientError.CurrentUserDoesNotExist)`

Reply:
fixed in a5002aa28109294765a23d548da3befb76427722